config/confighttp/compressor.go

```diff
@@ -22,36 +24,47 @@ type writeCloserReset interface {
 	Reset(w io.Writer)
 }
 
-var (
-	_          writeCloserReset = (*gzip.Writer)(nil)
-	gZipPool                    = &compressor{pool: sync.Pool{New: func() any { return gzip.NewWriter(nil) }}}
-	_          writeCloserReset = (*snappy.Writer)(nil)
-	snappyPool                  = &compressor{pool: sync.Pool{New: func() any { return snappy.NewBufferedWriter(nil) }}}
-	_          writeCloserReset = (*zstd.Encoder)(nil)
-	// Concurrency 1 disables async decoding via goroutines. This is useful to reduce memory usage and isn't a bottleneck for compression using sync.Pool.
-	zStdPool                  = &compressor{pool: sync.Pool{New: func() any { zw, _ := zstd.NewWriter(nil, zstd.WithEncoderConcurrency(1)); return zw }}}
-	_        writeCloserReset = (*zlib.Writer)(nil)
-	zLibPool                  = &compressor{pool: sync.Pool{New: func() any { return zlib.NewWriter(nil) }}}
-)
-
 type compressor struct {
 	pool sync.Pool
 }
 
+// Gets the compression type and level from the configuration.
+func getCompression(compressionField configcompression.Type) (compressionType configcompression.Type, compressionLevel int) {
+	parts := strings.Split(string(compressionField), "/")
+
+	compressionLevel = zlib.DefaultCompression
+	compressionType = configcompression.Type(parts[0])
+	if len(parts) > 1 {
+		levelStr := parts[1]
+		if level, err := strconv.Atoi(levelStr); err == nil {
```

Comment:
We should have return the error here instead of falling back to default. This should never happen though if we apply validation initially

Reply:
Falling back to the default compression level here, since that's whats used when there isn't any level specified.

Reply:
So in this case we'd be treating an invalid level as no level specified. Could this work as an alternative to rejecting compression config entirely? I chose this approach because level is intended to be optional.